packages/kbn-pm/src/utils/bazel/get_cache_folders.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * and the Server Side Public License, v 1; you may not use this file except in
+ * compliance with, at your election, the Elastic License or the Server Side
+ * Public License, v 1.
+ */
+
+import { dirname, resolve } from 'path';
+import { spawn } from '../child_process';
+
+async function rawRunBazelInfoRepoCache() {
+  const { stdout: bazelRepositoryCachePath } = await spawn('bazel', ['info', 'repository_cache'], {
+    stdio: 'pipe',
+  });
+  return bazelRepositoryCachePath;
+}
+
+export async function getBazelDiskCacheFolder() {
+  return resolve(dirname(await rawRunBazelInfoRepoCache()), 'disk-cache');
```

Comment:
Is `repository-cache` actually a file here and not a directory?

Reply:
I am leaning towards just using `REPO_ROOT` from `@kbn/utils` to construct these paths.

Reply:
Actually, we might be better off converting this to a paths.ts so they are constants.

Reply:
Is a directory but that path method works both on folders and files.

Reply:
Yeah, then I think it's weird to use `dirname` to treat it as a file. I think the recommendation of using consts with `REPO_ROOT` would be best.

Reply:
@tylersmalley the cache folder needs to be configured in a location outside the repo. Otherwise it will be created relative to the cwd where you are calling `bazel` from. Just checked with the Bazel team and that is the recommended way to do it for the moment (outside the repo) and thats why I need to apply that folder discovery (to work across every OS)